### PR TITLE
Adding QuestionAnsweringTask class to the question answering task

### LINKED ISSUE #537: Create QuestionAnsweringTask based on our SummarizationTask

### PR DESCRIPTION
**Pull request by karthikrangasai**
## What does this PR do?
The implementation of QuestionAnsweringTask class to the question_answering task.

Fixes #537 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/text/seq2seq/question_answering/test_model.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@pytest.mark.skipif(not _TEXT_TESTING, reason="text libraries aren't installed.")
+def test_init_train(tmpdir):
+    model = QuestionAnsweringTask(TEST_BACKBONE)
+    train_dl = torch.utils.data.DataLoader(DummyDataset())
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True)
+    trainer.fit(model, train_dl)
+
+
+@pytest.mark.skipif(not _TEXT_TESTING, reason="text libraries aren't installed.")
+def test_jit(tmpdir):
```
Comment (Borda) [inline tests/text/seq2seq/question_answering/test_model.py:58]: can you be more descriptive about the test?

Comment (ethanwharris) [reply tests/text/seq2seq/question_answering/test_model.py:58]: Haha, this is probably my fault, they are all just `test_jit` at the moment.

Comment (karthikrangasai) [reply tests/text/seq2seq/question_answering/test_model.py:58]: I used the tests from `tests/text/seq2seq/summarization/test_model.py` directly, but I believe that the test is to check for the compatibility with TorchScript. Should I rename it to `test_compatibility_with_jit` ?

Comment (ethanwharris) [reply tests/text/seq2seq/question_answering/test_model.py:58]: @karthikrangasai I think the best way forward would be if you want to open an issue to rename those tests then we could hava a separate PR that renames them all at once so we don't get inconsistent. I'll mark this is as resolved :smiley:

File: flash/text/seq2seq/question_answering/model.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import torch
+from torchmetrics import Metric
+
+from flash.text.seq2seq.core.metrics import RougeMetric
+from flash.text.seq2seq.core.model import Seq2SeqTask
+
+
+class QuestionAnsweringTask(Seq2SeqTask):
+    """The ``QuestionAnsweringTask`` is a :class:`~flash.Task` for Seq2Seq text question answering. For more details, see
+    :ref:`question_answering`.
```
Comment (ethanwharris) [inline flash/text/seq2seq/question_answering/model.py:25]:
```suggestion
    """The ``QuestionAnsweringTask`` is a :class:`~flash.Task` for Seq2Seq text question answering. For more details,
    see :ref:`question_answering`.
```

File: tests/text/seq2seq/question_answering/test_model.py (hunk 1 above)
Comment (karthikrangasai) [reply tests/text/seq2seq/question_answering/test_model.py:58]: @ethanwharris sure. Once the QuestionAnsweringTask is completed, I will work on that PR to keep the flow of code consistent throughout.
Should I open an issue now itself ?

File: flash/text/seq2seq/question_answering/model.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            num_beams=num_beams
+        )
+        self.rouge = RougeMetric(
+            rouge_newline_sep=rouge_newline_sep,
+            use_stemmer=use_stemmer,
+        )
+
+    @property
+    def task(self) -> str:
+        return "question_answering"
```
Comment (ethanwharris) [inline flash/text/seq2seq/question_answering/model.py:76]: I think this is just legacy so can be removed
```suggestion
```

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:

File: tests/text/seq2seq/question_answering/test_model.py (hunk 1 above)
Comment (ethanwharris) [reply tests/text/seq2seq/question_answering/test_model.py:58]: @karthikrangasai Yeah, you can go ahead and open an issue for it and I can assign you to it :smiley:

**Pull request merged**